ibis/backends/tests/test_generic.py

```diff
@@ -699,111 +699,6 @@ def test_select_filter_select(backend, alltypes, df):
-def test_bitwise_columns(backend, con, alltypes, df, op, left_fn, right_fn):
-    expr = op(left_fn(alltypes), right_fn(alltypes)).name("tmp")
-    result = con.execute(expr)
-
-    expected = op(left_fn(df), right_fn(df)).rename("tmp")
-    backend.assert_series_equal(result, expected)
-
-
-@pytest.mark.parametrize(
-    ("op", "left_fn", "right_fn"),
-    [
-        param(
-            lshift,
-            lambda t: t.int_col,
-            lambda t: t.int_col,
-            id="lshift_col_col",
-        ),
-        param(
-            lshift,
-            lambda _: 3,
-            lambda t: t.int_col,
-            marks=pytest.mark.broken(
-                ["impala"],
-                reason="impala's behavior differs from every other backend",
-            ),
-            id="lshift_scalar_col",
-        ),
-        param(lshift, lambda t: t.int_col, lambda _: 3, id="lshift_col_scalar"),
-        param(rshift, lambda t: t.int_col, lambda t: t.int_col, id="rshift_col_col"),
-        param(rshift, lambda _: 3, lambda t: t.int_col, id="rshift_scalar_col"),
-        param(rshift, lambda t: t.int_col, lambda _: 3, id="rshift_col_scalar"),
-    ],
-)
-@pytest.mark.notimpl(["bigquery", "dask", "datafusion", "pandas"])
-@pyspark_no_bitshift
-def test_bitwise_shift(backend, alltypes, df, op, left_fn, right_fn):
-    expr = op(left_fn(alltypes), right_fn(alltypes)).name("tmp")
-    result = expr.execute()
-
-    pandas_left = getattr(left := left_fn(df), "values", left)
-    pandas_right = getattr(right := right_fn(df), "values", right)
-    expected = pd.Series(
-        op(pandas_left, pandas_right),
-        name="tmp",
-        dtype="int64",
-    )
-    backend.assert_series_equal(result, expected)
-
-
-@pytest.mark.parametrize(
-    "op",
-    [
-        param(and_, marks=[pytest.mark.notimpl(["snowflake"])]),
-        param(or_, marks=[pytest.mark.notimpl(["snowflake"])]),
-        param(xor, marks=[pytest.mark.notimpl(["snowflake"])]),
-        param(lshift, marks=pyspark_no_bitshift),
-        param(rshift, marks=pyspark_no_bitshift),
-    ],
-)
-@pytest.mark.parametrize(
-    ("left", "right"),
-    [param(4, L(2), id="int_col"), param(L(4), 2, id="col_int")],
-)
-@pytest.mark.notimpl(["bigquery", "dask", "datafusion", "pandas"])
-def test_bitwise_scalars(con, op, left, right):
-    expr = op(left, right)
-    result = con.execute(expr)
-    expected = op(4, 2)
-    assert result == expected
-
-
-@pytest.mark.notimpl(["bigquery", "dask", "datafusion", "pandas", "snowflake"])
-def test_bitwise_not_scalar(con):
-    expr = ~L(2)
-    result = con.execute(expr)
-    expected = -3
-    assert result == expected
-
-
-@pytest.mark.notimpl(["bigquery", "dask", "datafusion", "pandas", "snowflake"])
-def test_bitwise_not_col(backend, alltypes, df):
-    expr = (~alltypes.int_col).name("tmp")
-    result = expr.execute()
-    expected = ~df.int_col
-    backend.assert_series_equal(result, expected.rename("tmp"))
-
-
-@pytest.mark.notimpl(
```

Comment:
Now the result of the operation is as below. I'm not sure if this is expected behavior or if we should throw an exception.
```
Translation to backend failed
Error message: No translation rule for <class 'ibis.expr.operations.temporal.TimestampDiff'>
Expression repr follows:
r0 := AlchemyTable: functional_alltypes
  index           int64
  Unnamed: 0      int64
  id              int64
  bool_col        boolean
  tinyint_col     int64
  smallint_col    int64
  int_col         int64
  bigint_col      int64
  float_col       float64
  double_col      float64
  date_string_col string
  string_col      string
  timestamp_col   timestamp
  year            int64
  month           int64

Selection[r0]
  selections:
    r0
    str_col:             NullIf(StringReplace(r0.string_col, pattern='1', replacement=''), null_if_expr='2')
    date_col:            Date(r0.timestamp_col)
    delta_col:           TimestampNow() - r0.timestamp_col
```

Reply:
Any particular reason you removed this marker?

Reply:
Oh, I see now. This behavior is expected, but not ideal. Ideally we should implement some form of `TimestampDiff` for snowflake, but that can be done in a follow up.

Reply:
Now this tests does not raise exception, so when we have market with `strict` set to True set, this test fails.

> [XPASS(strict)] snowflake doesn't implement timestamp subtraction

https://github.com/ibis-project/ibis/actions/runs/3885821095/jobs/6630175834